azure/azure-release-template.yml

```diff
@@ -65,62 +69,74 @@ extends:
 
           if [[ "$APIGEE_ENVIRONMENT" == *"sandbox"* ]]; then
-            export PACT_PROVIDER=nhsd-apim-eps-sandbox
+          export PACT_PROVIDER=nhsd-apim-eps-sandbox
           else
-            export PACT_PROVIDER=nhsd-apim-eps
-            if [ "$APIGEE_ENVIRONMENT" == "int" ]
-            then
+          export PACT_PROVIDER=nhsd-apim-eps
+          if [ "$APIGEE_ENVIRONMENT" == "int" ]
+          then
               export IDP_URL="https://nhsd-apim-testing-$APIGEE_ENVIRONMENT-ns.herokuapp.com"
               docker pull artronics/nhsd-login-docker:latest > /dev/null
               export APIGEE_ACCESS_TOKEN=$(docker run --rm artronics/nhsd-login-docker:latest "$IDP_URL")
-            elif [ "$APIGEE_ENVIRONMENT" == "ref" ]
-            then
+          elif [ "$APIGEE_ENVIRONMENT" == "ref" ]
+          then
               export IDP_URL="$(REF_IDP_URL)"
               export AUTH_BEARER_TOKEN="$(REF_AUTH_BEARER_TOKEN)"
               docker pull booshi/nhsd-login-docker:latest > /dev/null
-              export APIGEE_ACCESS_TOKEN=$(docker run --rm -e AUTH_BEARER_TOKEN=$AUTH_BEARER_TOKEN booshi/nhsd-login-docker:latest "$IDP_URL")
-            else
+              export APIGEE_ACCESS_TOKEN=$(docker run --rm -e AUTH_BEARER_TOKEN="$AUTH_BEARER_TOKEN" booshi/nhsd-login-docker:latest "$IDP_URL")
+          else
               export IDP_URL="https://nhsd-apim-testing-$APIGEE_ENVIRONMENT.herokuapp.com"
               docker pull artronics/nhsd-login-docker:latest > /dev/null
               export APIGEE_ACCESS_TOKEN=$(docker run --rm artronics/nhsd-login-docker:latest "$IDP_URL")
-            fi
+          fi
           fi
 
           # Publish
-          cd $SERVICE_NAME/$SERVICE_ARTIFACT_NAME/pact
+          cd "$SERVICE_NAME"/"$SERVICE_ARTIFACT_NAME"/pact
           rm -rf node_modules && npm install > /dev/null
           make create-pacts > /dev/null
           chmod +x ./broker/publish.ts
           make publish-pacts > /dev/null
 
           # Poll deploying API until our deployed version matches the release version
-          url="https://$APIGEE_ENVIRONMENT.api.service.nhs.uk/$SERVICE_BASE_PATH/_ping"
+          url="https://$APIGEE_ENVIRONMENT.api.service.nhs.uk/$SERVICE_BASE_PATH/_status"
           interval_in_seconds=5
           releaseCommit="$(Build.SourceVersion)"
-          path=".commitId"
-          printf "\nPolling '$url' every $interval_in_seconds seconds, until commit is: '$releaseCommit'\n"
-          while true;
+          printf "\nPolling %s every %s seconds, until commit is: %s\n" "$url" "$interval_in_seconds" "$releaseCommit"
+          attempts=0
+          success=0
+          until [ $attempts -eq 15 ]
           do
-            deployedCommit=`curl $url -s | jq -r $path`
-            if [[ "$deployedCommit" == "$releaseCommit" ]]; then
+          responseData=$(curl -H "apiKey: $(status-endpoint-api-key)" "$url" -s)
+          deployedCommit=$(echo $responseData | jq -r ".commitId")
```

Comment:
What happens if the coordinator isn't up yet and so `responseData` is empty?

Reply:
Not 'nicely' really. Curl returns a load of crap and then JQ fails to extract the expected fields